addons/web/static/src/views/calendar/hooks.js

```diff
@@ -114,13 +115,20 @@ export function useFullCalendar(refName, params) {
         }
     });
 
-    let isWeekendVisible = params.isWeekendVisible;
+    let isWeekendVisible = params.weekends;
     onWillUpdateProps((np) => {
         isWeekendVisible = np.isWeekendVisible;
     });
     onPatched(() => {
         instance.refetchEvents();
         instance.setOption("weekends", isWeekendVisible);
+        if (params.weekNumbers && !params.allDaySlot) {
+            const elements = document.querySelectorAll(".fc-dayGridMonth-view");
+            const weekText = instance.currentData.options.weekText;
+            elements.forEach(el => {
+                makeWeekColumn({ el, weekText });
+            });
+        }
```

Comment:
I'm not sure to understand that part. And it probably requires a test. @mcm-odoo could you review this? Thanks

Reply:
>I'm not sure to understand that part.

It has been added to create a seperate week number column when we toggle `Show Weekends` from scale selection.
Previously in v4 of fullcalendar there was a option called weekNumbersWithinDays [docs](https://fullcalendar.io/docs/v4/weekNumbersWithinDays) which was used to create a seperate week number column. 

But currently in v6 we don't have that option, so for it's alternative `makeWeekColumn` function is added - [PR](https://github.com/odoo/odoo/pull/150097). We patch the Fullcalendar at the end `viewDidMount` of the "Preact Lifecycle" to add the week column. But when we toggle `Show Weekends` from scale selection `viewDidMount` hook is not called due to which a seperate week column is not created.

cc: @rfr-odoo

Reply:
I'm missing something but can't we use the `get customOptions()` ?
It seems odd to have to use the two API (ours `customOptions` and the `setOption` of FC ) in the same time.
I don't like that, but as it's a FIX it's ok (i think), but in master it would be better to not have to interact with FC directly but instead relie on our API and if we need to add a setter to `options` and `customOptions` or something else why not.

cc: @mcm-odoo

Reply:
Hello @rfr-odoo , @mcm-odoo 

Actually I have used `setOption` to get values in `params` of hooks.js when useFullCalendar is called. 
https://github.com/odoo/odoo/blob/d275db688523a90c71130ef24d306bc39d6fd7d8/addons/web/static/src/views/calendar/calendar_year/calendar_year_renderer.js#L30-L33
We are passing `this.getOptionsForMonth(month)` as params. And in `getOptionsForMonth(month)` we are returning only options (no customOptions) -
https://github.com/odoo/odoo/blob/d275db688523a90c71130ef24d306bc39d6fd7d8/addons/web/static/src/views/calendar/calendar_year/calendar_year_renderer.js#L114-L119
But as per your suggestion If I add `weekends: this.props.isWeekendVisible,` in `set customOptions()`, I have to also return customOptions as params to useFullCalendar. For example -
```
getOptionsForMonth(month) {
        return {
            ...this.options,
            ...this.customOptions,
            initialDate: this.getDateWithMonth(month),
        };
    }
```
Please let me know your suggestions and correct me if I am missing something.

And also the main issue of the task is occuring because `makeWeekColumn` is called in `viewDidMount` hook but when we toggle `Show Weekends` from scale selection the `viewDidMount` is not encountered due to which a seperate week column is not created.

Thanks!!

Reply:
```javascript
getOptionsForMonth(month) {
  return {
    ...this.options,
    ...this.customOptions,
    initialDate: this.getDateWithMonth(month),
  };
}
```
will generate a warning (or error) in the console, FC don't like to have unknown options.

An other solution can be to force the redraw using (to test if it works)
```
calendar.destroy();
calendar.render();
```

Reply:
Re-rendering the calendar seems to be working :+1: 
But now in onPatched hook over here - 
```
    onPatched(() => {
        instance.refetchEvents();
        instance.setOption("weekends", isWeekendVisible);
        instance.destroy();
        instance.render();
    });
```
is it still require to have `instance.refetchEvents();` after the rerendering of calendar?